contrib/epee/include/storages/parserse_base_utils.h

```diff
@@ -304,25 +133,7 @@ namespace misc_utils
           return false;
         }
       }
-      inline void match_word2(std::string::const_iterator& star_end_string, std::string::const_iterator buf_end, boost::string_ref& val)
-      {
-        val.clear();
-
-        for(std::string::const_iterator it = star_end_string;it != buf_end;it++)
-        {
-          if (!(lut[(uint8_t)*it] & 4))
-          {
-            val = boost::string_ref(&*star_end_string, std::distance(star_end_string, it));
-            if(val.size())
-            {
-              star_end_string = --it;
-              return;
-            }else 
-              ASSERT_MES_AND_THROW("failed to match word number in json entry: " << std::string(star_end_string, buf_end));
-          }
-        }
-        ASSERT_MES_AND_THROW("failed to match word number in json entry: " << std::string(star_end_string, buf_end));
-      }
+      void match_word2(std::string::const_iterator& star_end_string, std::string::const_iterator buf_end, boost::string_ref& val);
       inline bool match_word(std::string::const_iterator& star_end_string, std::string::const_iterator buf_end, boost::string_ref& val)
```

Comment:
Why don't move all the remaining code out of the header as well?

Reply:
Because:
1) it doesn't hurt, as they impose no additional dependencies in the header list (my PoV),
2) the runtime performance is slightly improved when all that such a trivial call does is entering and redirecting the parameters further (both my and @vtnerd 's PoV).

Reply:
This `try`/`catch` block is little different since it has to generate a "landing pad" for the catch, but seems fine to leave this `inline`d. The compiler can still elect to _not_ inline the code.